capstone/src/main/webapp/script.js

```diff
@@ -31,14 +32,150 @@ async function submitInput() {
+        return;
+      }
+      const toxicityScore = response.attributeScores.TOXICITY.summaryScore.value;
+      var attributes = [];
+      Object.keys(response.attributeScores).forEach((attribute) => {
+        attributes.push([attribute, response.attributeScores[attribute].summaryScore.value]);
+      });
+      attributes.sort(function(a, b) {
+        return b[1] - a[1];
+      });
+
+      // Color the substring appropriately	
+      const substringEl = createAnyElement('span', substrings[i]);
+      substringEl.className = 'green-background segment';
+      if (toxicityScore >= 0.8) {
+        substringEl.className = 'red-background segment';
+      } else if (toxicityScore >= 0.2) {
+        substringEl.className = 'yellow-background segment';
+      }
+     
+      // Attach a tooltip (info-box for the substring)
+      const tooltipEl = createTooltip(attributes);
+      substringEl.appendChild(tooltipEl);
+      result.appendChild(substringEl);
+    }
+  }
+  analysisContainer.removeChild(loadingEl);
+  analysisContainer.appendChild(result);
+}
+
+/** Breaks up a string into its words or sentences and puts the substrings in an array */
+function getSubstrings(text, delimiter) {
+  if (delimiter === ' ') {
+    return text.match(/\S+\s*/g);  // Regular expression for getting words
+  } else {
+    return text.match(/([^\.!\?]+[\.!\?]+)|([^\.!\?]+$)/g);  // Regular expression for getting sentences
+  }
+}
+
+/** Creates a tooltip for a substring in the detailed analysis */
+function createTooltip(attributes) {
+  const tooltipEl = document.createElement('div');
+  const headerEl = document.createElement('div');
+  const titleInfoEl = document.createElement('div');
+  const imageEl = document.createElement('img');
+  const bodyEl = document.createElement('div');
+  const titleEl = createAnyElement('h3', 'Perspective Feedback');
+  const subtitleEl = createAnyElement('p', 'based on selected attributes');
+
+  tooltipEl.className = 'tooltip';      
+  headerEl.className = 'header';
+  titleInfoEl.className = 'title-info';
+  titleEl.className = 'tooltip-title';
+  subtitleEl.className = 'tooltip-subtitle';
+  bodyEl.className = 'tooltip-body';
+
+  imageEl.setAttribute('src', 'assets/apple-touch-icon.png');
+  imageEl.setAttribute('alt', 'Perspective Logo');
+
+  titleInfoEl.appendChild(titleEl);
+  titleInfoEl.appendChild(subtitleEl);
+  headerEl.appendChild(imageEl);
+  headerEl.appendChild(titleInfoEl);
+  tooltipEl.appendChild(headerEl);
+  tooltipEl.appendChild(bodyEl);
+
+  for (const attribute of attributes) {
+    bodyEl.appendChild(createAnyElement('p', attribute[0] + ': ' + decimalToPercentage(attribute[1])));
+  }
+  return tooltipEl;
+}
+
+/** Converts decimals to percentages */
+function decimalToPercentage(decimal) {
+  const decimalAsString = decimal.toString();
+  return decimalAsString.slice(2, 4) + '.' + decimalAsString.slice(4, 6) + '%';
```

Comment:
Does `decimal * 100 + '%'` work?

Reply:
Yup, I also replaced it with `(decimal * 100).toFixed(2) + '%'` to round to the hundredths place